test/runnable/sdtor.d

```diff
@@ -3899,20 +3900,22 @@ bool test14022()
         t.sb[1].x = 'y';
         assert(sa == [S('a'), S('b'), S('c')]);
         assert(t.sb == [S('x'), S('y')]);
-        assert(op == "BC");
+        // Array literals destroyed at the end of scope
+        assert(op == "BCcbcbayx");
     }
-    assert(op == "BCyxcba");
+    assert(op == "BCcbcbayxyxcba");
 
     op = null;
     {
         S[3] sx = [S('a'), S('b'), S('c')];
         T t;    t.sb[0].x = 'x';
-                t.sb[1].x = 'y';
+        t.sb[1].x = 'y';
         t.test(sx[1..3]);
         assert(op == "BxCy");
         assert(t.sb == [S('b'), S('c')]);
+        assert(op == "BxCycb");
     }
-    assert(op == "BxCycbcba");
```

Comment:
This is a significant behavior change. Needs more documentation than just changing the test suite. Needs entry in changelog.